include/swift/Parse/Parser.h

```diff
@@ -158,6 +158,11 @@ class Parser {
   bool InSwiftKeyPath = false;
   bool InFreestandingMacroArgument = false;
 
+#if SWIFT_BUILD_SWIFT_SYNTAX
+  // This Parser is a fallback parser for ASTGen.
+  bool IsForASTGen = false;
```

Comment:
Why did you choose to put this behind a `#if`? Seems like a potential source of issues in the future if you use it from somewhere that isn’t guarded by `SWIFT_BUILD_SWIFT_SYNTAX` and that you won’t see locally if you have `SWIFT_BUILD_SWIFT_SYNTAX` enabled.

Reply:
I don't think "we" (developers with `SWIFT_BUILD_SWIFT_SYNTAX ` environment) would do much in `#else` branches in Parser.  I'm more worried that developers for non-`SWIFT_BUILD_SWIFT_SYNTAX` environments use `IsForASTGen` mistakingly.

Reply:
What do you imagine someone might use `IsForASTGen` for that would be guarded by the `#if`?

Reply:
I have no idea. But if it's inside `#if SWIFT_BUILD_SWIFT_SYNTAX` they cannot use it by mistake. So I thought it's safer to guard it.
Though I'm totally OK not to guard this, if you think that's better.

Reply:
It’s fine. If you think it provides value, let’s guard it.

Reply:
FWIW we have had issues in the past where `#if`'ing out fields leads to crashes if the library and its client ever disagree about the condition (IIRC most recently we had issues with `#ifndef NDEBUG`). Not sure if that can be the case here, but I would tend on the side of not `#if`'ing out fields in general.